src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp

```diff
@@ -71,44 +72,28 @@ void FlybyPoint::UpdateState(uint64_t CurrentSimNanos)
         if (this->firstRead) {
             this->firstRead = false;
         }
-
-        /*! read and allocate the attitude navigation message */
-        NavTransMsgPayload relativeState = this->filterInMsg();
-
         /*! compute velocity/radius ratio at time of read */
-        this->f0 = v3Norm(relativeState.v_BN_N) / v3Norm(relativeState.r_BN_N);
+        this->r_BN_N = Eigen::Map<Eigen::Vector3d>(relativeState.r_BN_N);
+        this->v_BN_N = Eigen::Map<Eigen::Vector3d>(relativeState.v_BN_N);
+        this->f0 = this->v_BN_N.norm() / this->r_BN_N.norm();
 
         /*! compute radial (ur_N), velocity (uv_N), along-track (ut_N), and out-of-plane (uh_N) unit direction vectors */
-        double ur_N[3];
-        v3Normalize(relativeState.r_BN_N, ur_N);
-        double uv_N[3];
-        v3Normalize(relativeState.v_BN_N, uv_N);
-        double uh_N[3];
-        double ut_N[3];
-        if (1 - v3Dot(ur_N, uv_N) < this->epsilon) {
-            v3Perpendicular(ur_N, uh_N);
-            v3Normalize(uh_N, uh_N);
-            v3Cross(uh_N, ur_N, ut_N);
-            // compute flight path angle at the time of read
-            this->gamma0 = M_PI_2;
-            this->singularityFlag = plusInfinity;
-        }
-        else if (v3Dot(ur_N, uv_N) + 1 < this->epsilon) {
-            v3Perpendicular(ur_N, uh_N);
-            v3Normalize(uh_N, uh_N);
-            v3Cross(uh_N, ur_N, ut_N);
-            // compute flight path angle at the time of read
-            this->gamma0 = -M_PI_2;
-            this->singularityFlag = minusInfinity;
-        }
-        else {
-            v3Cross(ur_N, uv_N, uh_N);
-            v3Normalize(uh_N, uh_N);
-            v3Cross(uh_N, ur_N, ut_N);
-            // compute flight path angle at the time of read
-            this->gamma0 = atan(v3Dot(relativeState.v_BN_N, ur_N) / v3Dot(relativeState.v_BN_N, ut_N));
-            this->singularityFlag = nonSingular;
-        }
+        Eigen::Vector3d ur_N = this->r_BN_N / this->r_BN_N.norm();
+        Eigen::Vector3d uv_N = this->v_BN_N / this->v_BN_N.norm();
+        /*! assert r and v are not collinear (collision trajectory) */
+        assert(std:abs(1 - ur_N.dot(uv_N)) < this->toleranceForCollinearity);
+
+        Eigen::Vector3d uh_N;
```

Comment:
:large_orange_diamond: Is there a preference for declaring and assigning the variable on different lines? I don't know if there's a benefit either way. Just curious.

Reply:
improved this too